src/praisonai-agents/praisonaiagents/gateway/protocols.py

```
@@ -689,6 +689,126 @@ async def purge_acknowledged(self, max_age_seconds: int = 86400) -> int:
+            key: The entry key returned by enqueue()
+            error: Error description
+            permanent: If True, won't retry this message
+            
+        Returns:
+            True if marked successfully, False if not found
+        """
+        ...
+    
+    async def drain(
+        self,
+        sender: Callable[[str, Dict[str, Any]], Awaitable[bool]],
```

Ruff (0.15.17)

[error] Undefined name `Awaitable`

(F821)